### PR TITLE
Added check for fertility pills, including broodmother

### PR DESCRIPTION
**Pull request by AceXP**
- What is the purpose of the pull request?

Special NPC's (like Lilaya and Kate) should treat brood mother pills much the same as other fertility pills.

- Give a brief description of what you changed or added.

Added a isTypeOneOf method to AbstractItem, and use this when determining if a special reaction is needed.

Also, Roxy now sells brood mother pills, and there is a chance random NPC's will have them as well.

And several scenes still mention the Breeder pills as pink, but they have been purple for some time. Those occurances have been changed.

- Are any new graphical assets required?

No

- Has this change been tested? If so, mention the version number that the test was based on.

Yes, tested with 3.9.9

- So we have a better idea of who you are, what is your Discord Handle?

AceXP#0930

### CODE REVIEW

File: src/com/lilithsthrone/game/inventory/item/AbstractItem.java [hunk 1]
```diff
@@ -282,6 +283,11 @@ public boolean isGift() {
 		return itemType.isGift();
 	}
 
+	public boolean isFertilityPill() {
+		return this.getItemType().equals(ItemType.getItemTypeFromId("innoxia_pills_fertility"))
+				|| this.getItemType().equals(ItemType.getItemTypeFromId("innoxia_pills_broodmother"));
+	}
+
```
Comment (CognitiveMist) [inline src/com/lilithsthrone/game/inventory/item/AbstractItem.java:290]: I wonder if a more generic `item.isTypeOneOf(String itemType...)` would be better. More useful for modders, I think.

File: src/com/lilithsthrone/game/inventory/item/AbstractItem.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import java.util.List;
 import java.util.Set;
 
+import com.lilithsthrone.game.inventory.enchanting.TFModifier;
```
Comment (CognitiveMist) [inline src/com/lilithsthrone/game/inventory/item/AbstractItem.java:8]: Extra?

File: src/com/lilithsthrone/game/inventory/item/AbstractItem.java (hunk 1 above)
Comment (AceXP) [reply src/com/lilithsthrone/game/inventory/item/AbstractItem.java:289]: Done

File: src/com/lilithsthrone/game/inventory/item/AbstractItem.java (hunk 2 above)
Comment (AceXP) [reply src/com/lilithsthrone/game/inventory/item/AbstractItem.java]: And gone

**Review by CognitiveMist — APPROVED** [review]

**Comment by Innoxia** [comment]
Thank you for this!

**Pull request merged**